### DISCUSSION TITLE
initial setup

### DISCUSSION THREAD

**Discussion by cipitaua** [v1993/linuxmotehook2#2] [Q&A] [answered]
Hello and thank you for this great piece of code :)

I would like to test it but I could not find a list of possible options to be placed in the .ini config file. Could you please quickly explain how to set it up? Also, is there any option to increase verbosity?

Thanks again

**Comment by v1993** [comment, marked as answer]
[Example config file](https://github.com/v1993/linuxmotehook2/blob/main/ExampleConfig.ini) includes all applicable options. Section name is MAC of you Wiimote (you can run with an empty config to get MAC or look it up in bluetooth settings). I'm going to make a proper documentation for config file as well as application to help with creating it once I'll get back to working on this program (it's now a lot less stressful with CI in place).

As for increasing verbosity, you can set `G_MESSAGES_DEBUG=linuxmotehook2`, but note that extra information might be a bit too technical sometimes.

If you have any further questions about options, feel free to ask!

**Reply by cipitaua** [reply]
thanks for the prompt reply. About the `Orientation` flag, can it only be `normal` or `sideways-left`? And how to set the several calibration values?

**Reply by v1993** [reply]
`normal`, `sideways-left` and `inverted` are those likely to be useful. Calibration values would require additional program, which is not implemented yet. Gyro calibration can be obtained from original linuxmotehook's autocalibration script, which should still work despite issues with main program.

By the way - any option can be omitted, in which case it will use a reasonable default value. Those in per-device sections override global options as expected.

What emulator are you trying to use btw?

**Reply by cipitaua** [reply]
mainly cemu

I'll try and give you a feedback asap :)

**Reply by v1993** [reply]
I've just pushed a new commit - it should help with avoiding gyro drift even without calibration values (please note that it's not in (testing) PPA yet - that's still building for now).

**Reply by cipitaua** [reply]
I presently have the following issue:
```
$ sudo apt update
...
W: GPG error: https://ppa.launchpadcontent.net/v19930312/linuxmotehook2/ubuntu impish InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 4912548893EFE535
E: The repository 'https://ppa.launchpadcontent.net/v19930312/linuxmotehook2/ubuntu impish InRelease' is not signed.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
```
and

```
$ sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 4912548893EFE535
Warning: apt-key is deprecated. Manage keyring files in trusted.gpg.d instead (see apt-key(8)).
Executing: /tmp/apt-key-gpghome.bNYYcKIkiX/gpg.1.sh --keyserver keyserver.ubuntu.com --recv-keys 4912548893EFE535
gpg: keyserver receive failed: Server indicated a failure
```

**Reply by v1993** [reply]
How did you add PPA? I have no idea what could the issue be here aside from temporary keyserver failure.

**Reply by cipitaua** [reply]
don't know. I've circumvented by placing
```
deb [allow-insecure=yes allow-downgrade-to-insecure=yes] https://ppa.launchpadcontent.net/v19930312/linuxmotehook2/ubuntu ./impish main
```
in my /etc/apt/sources.list.d/linuxmotehook2.list

**Reply by cipitaua** [reply]
update: I've tried again to import the repository key, and I get a different error:

```
$ sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 4912548893EFE535
Warning: apt-key is deprecated. Manage keyring files in trusted.gpg.d instead (see apt-key(8)).
Executing: /tmp/apt-key-gpghome.hITGGAOLQh/gpg.1.sh --keyserver keyserver.ubuntu.com --recv-keys 4912548893EFE535
gpg: key 4912548893EFE535: public key "Launchpad PPA for Val" imported
gpg: Total number processed: 1
gpg:               imported: 1
gpg: invalid key resource URL '/tmp/apt-key-gpghome.hITGGAOLQh/home:ungoogled_chromium.asc.gpg'
gpg: keyblock resource '(null)': General error
gpg: key 76F1A20FF987672F: 1 signature not checked due to a missing key
gpg: key 0F92290A445B9007: 6 signatures not checked due to missing keys
gpg: key 02456C79B2FD48BF: 1 signature not checked due to a missing key
gpg: key 02456C79B2FD48BF: 1 signature not checked due to a missing key
gpg: key D94AA3F0EFE21092: 3 signatures not checked due to missing keys
gpg: key 871920D1991BC93C: 1 signature not checked due to a missing key
gpg: Total number processed: 29
gpg:       skipped new keys: 29

```

**Reply by v1993** [reply]
This is unrelated to my repo. Seek help with this one elsewhere.

**Comment by cipitaua** [comment]
it seems to work, but I get two separate DSU controllers, one for the wiimote and the other for the nunchuck. I cannot figure a way to merge buttons into a single controllers. I'm using the ExampleConfig.ini with the revised mac address. Moreover, nunchuck buttons seems to work in PadTest, but not in cemu. In yuzu, by selecting input from 'any' I can map wiimote+nunchuck into pro controller, but only because yuzu has such feature.

**Reply by cipitaua** [reply]
by using moltengamepad *after* linuxmotehook2 I could merge the buttons into a single controller which is correctly recognized in cemu.

I need two apps to do one thing, but I'm fine with that :)

**Reply by v1993** [reply]
You shouldn't be using options from ExampleConfig.ini, because those calibrations are for my wiimote/nunchuck. Get rid of values that you haven't calibrated yourself and trust the defaults.

Wiimote and Nunchuck are separate devices by design and it won't change. Not only they are separate physically, it makes little sense to report nunchuck's motion via its separate device and its buttons via wiimotes core device. Not to mention that classic controller has too many buttons to report everything via single device, making extension button reporting inconsistent in that case.

Moltengamepad does not appear to interact with linuxmotehook2 in any manner, instead doing its own thing.

Not sure what's with nunchuck buttons not working in cemu. Don't think I have this problem (too busy to test rn).
